src/daft-logical-plan/src/optimization/rules/reorder_joins/join_graph.rs

```diff
@@ -221,18 +222,123 @@ impl JoinGraph {
         }
     }
 
+    fn apply_projections_and_filters_to_plan_builder(
+        &mut self,
+        mut plan_builder: LogicalPlanBuilder,
+    ) -> DaftResult<LogicalPlanBuilder> {
+        // Apply projections and filters in post-traversal order.
+        let mut reversed_items = self
+            .final_projections_and_filters
+            .drain(..)
+            .rev()
+            .peekable();
+        while let Some(projection_or_filter) = reversed_items.next() {
+            let is_last = reversed_items.peek().is_none();
+
+            match projection_or_filter {
+                ProjectionOrFilter::Projection(projections) => {
+                    if is_last {
+                        // The final projection is the output projection, so here we select the final projection.
+                        plan_builder = plan_builder.select(projections.clone())?;
+                    } else {
+                        // Intermediate projections might only transform a subset of columns, so we use `with_columns()` instead of `select()`.
+                        plan_builder = plan_builder.with_columns(projections.clone())?;
+                    }
+                }
+                ProjectionOrFilter::Filter(predicate) => {
+                    plan_builder = plan_builder.filter(predicate.clone())?;
+                }
+            }
+        }
+        Ok(plan_builder)
+    }
+
+    /// Converts a `JoinOrderTree` into a tree of inner joins.
+    /// Returns a tuple of the logical plan builder consisting of joins, and a bitmask indicating the plan IDs
+    /// that are contained within the current logical plan builder. The bitmask is used for determining join
+    /// conditions to use when logical plan builders are joined together.
+    fn build_joins_from_join_order(
```

Comment:
Spoke offline, put join conditions into JoinOrderTree instead since this should just be a 1:1 translation